paper_framework_sources/p1897_initial_executor_algorithms_for_cpp23.md

```diff
@@ -25,56 +35,71 @@ toc: false
+Once we have basic primitives, we can describe default implementations of the parallel algorithms, as well as `std::async`, in terms of these.
+
 In the long run we expect to have a much wider set of algorithms, potentially covering the full set in the current C++20 parallel algorithms.
 The precise customization of these algorithms is open to discussion: they may be individually customized and individually defaulted, or they may be optionally individually customized but defaulted in a tree such that customizing one is known to accelerate dependencies.
 It is open to discussion how we achieve this and that is an independent topic, beyond the scope of this paper.
 
 ## Summary
-Starting with [@P0443R11] as a baseline we have the following customization points:
+Starting with [@P0443R13] as a baseline we have the following customization points:
 
- * `execute(executor, invocable) -> void`
+ * `connect(sender, receiver) -> operation_state`
+ * `start(operation_state) -> void`
  * `submit(sender, receiver) -> void`
  * `schedule(scheduler) -> sender`
+ * `execute(executor, invocable) -> void`
  * `set_done`
  * `set_error`
  * `set_value`
 
 and the following Concepts:
 
- * `executor`
  * `scheduler`
- * `callback_signal`
- * `callback`
+ * `receiver`
+ * `receiver_of`
  * `sender`
+ * `sender_to`
+ * `typed_sender`
+ * `operation_state`
+ * `executor`
+ * `executor_of`
 
 We propose immediately discussing the addition of the following algorithms:
 
  * `just(v)`
    * returns a `sender` of the value `v`
- * `just_via(sch, v)`
-   * a variant of the above that embeds the `via` algorithm
- * `via(s, sch)`
+ * `just_on(sch, v)`
+   * a variant of the above that embeds the `on` algorithm
+ * `on(s, sch)`
    * returns a sender that propagates the value or error from `s` on `sch`'s execution context
  * `sync_wait(s)`
    * blocks and returns the value type of the sender, throwing on error
  * `when_all(s...)`
    * returns a sender that completes when all Senders `s...` complete, propagating all values
- * `indexed_for(s, policy, rng, f)`
-   * returns a sender that applies `f` for each element of `rng` passing that element and the values from the incoming sender, completes when all `f`s complete propagating s's values onwards
+ <!--* `indexed_for(s, policy, rng, f)`
+   * returns a sender that applies `f` for each element of `rng` passing that element and the values from the incoming sender, completes when all `f`s complete propagating s's values onwards-->
  * `transform(s, f)`
    * returns a sender that applies `f` to the value passed by `s`, or propagates errors or cancellation
- * `bulk_transform(s, f)`
-   * returns a sender that applies `f` to each element in a range sent by `s`, or propagates errors or cancellation
+ <!--* `bulk_transform(s, f)`
+   * returns a sender that applies `f` to each element in a range sent by `s`, or propagates errors or cancellation-->
  * `handle_error(s, f)`
```

Comment:
I prefer to name this transform_error. I expect to have transform_value and transform_done - perhaps even in this base set. I added transform_done to libunifex already because I needed it.

Reply:
I suspect people will be concerned with the name "transform" anyway because the one in the standard already is elementwise on a range.

The other reason I kept this distinct is because it is more like let than transform in behaviour. In that sense I considered "let_error" as a name.

But in the end, I don't think we should worry too much about names at this point.

Reply:
I am not set on a particular name in std. I expect all these names to change. I would like for similar things to have similar names now so that those unfamiliar will be able to associate them.

Reply:
What would you do with "transform" that doesn't return a sender from its function?

If we keep that, then let_error would make more sense to me for consistency.

Reply:
I do not mind that they are similarly named. transform_value is a superset of transform, or transform is an alias to transform_value that always returns just(f(an...))

Reply:
Renamed to let_error along with let renaming to let_value as discussed.